src/ape_pm/compiler.py

```diff
@@ -19,20 +20,27 @@ def get_versions(self, all_paths: List[Path]) -> Set[str]:
         #       ``compilers`` field. You should not do this with a real compiler plugin.
         return set()
 
-    def compile(self, filepaths: List[Path]) -> List[ContractType]:
+    def compile(
+        self, filepaths: List[Path], base_path: Optional[Path] = None
+    ) -> List[ContractType]:
         contract_types: List[ContractType] = []
         for path in filepaths:
             with path.open() as f:
                 data = json.load(f)
 
+            source_id = (
+                str(get_relative_path(path, base_path))
+                if base_path and path.is_absolute()
+                else str(path)
+            )
```

Comment:
so this will need to be upstreamed to `ape-solidity` and `ape-vyper`?

Reply:
Yes, along with the new API parameter